src/MurmurClient.ts

```diff
@@ -0,0 +1,210 @@
+      });
+
+      // Extract the Set-Cookie header
+      const setCookieHeader = response.headers["set-cookie"];
+      if (setCookieHeader) {
+        // Store the cookies in the Axios instance's default headers to keep the session
+        this.http.defaults.headers.Cookie = setCookieHeader.join("; ");
+      }
+
+      return response.data;
+    } catch (error) {
+      throw new Error(`Authenticattion failed: ${error}`);
+    }
+  }
+
+  /**
+   * Creates a new wallet with a specified validity period.
+   *
+   * @param validity - The number of blocks in which the wallet will be valid.
+   * @param callback - The callback function to be called when the transaction is finalized.
+   * @returns A promise that resolves to a string indicating the block hash of the block in which
+   * the transaction was executed, or in which was included if it didn't execute.
+   */
+  async new(
+    validity: number,
+    callback: (result: any) => Promise<void> = async () => {}
+  ): Promise<void> {
+    const MAX_U32 = 2 ** 32 - 1;
+    if (!Number.isInteger(validity)) {
+      throw new Error("The validity parameter must be an integer.");
+    }
+
+    if (validity < 0 || validity > MAX_U32) {
+      throw new Error(
+        `The validity parameter must be within the range of 0 to ${MAX_U32}.`
+      );
+    }
+    const request: NewRequest = {
+      validity,
+      current_block: (await this.getCurrentBlock()).toNumber(),
+      round_pubkey: (await this.getRoundPublic()).toString(),
+    };
+
+    try {
+      const response = (await this.http.post("/create", request))
+        .data as CreateResponse;
+
+      const extrinsic = this.constructExtrinsic(response.payload);
+
+      this.submitExtrinsic(extrinsic, callback);
+
+      return Promise.resolve();
+    } catch (error) {
+      throw new Error(`New failed: ${error}`);
+    }
+  }
+
+  /**
+   * Executes a transaction to send a specified amount of tokens to a destination account.
+   *
+   * @param extrinsic - A submittable extrinsic.
+   * @param callback - The callback function to be called when the transaction is finalized.
+   * @returns A promise that resolves to a string indicating the result of the transaction.
+   */
+  async execute(
+    extrinsic: Extrinsic,
+    callback: (result: any) => Promise<void> = async () => {}
+  ): Promise<void> {
+    const request: ExecuteRequest = {
+      runtime_call: this.encodeExtrinsic(extrinsic),
+      current_block: (await this.getCurrentBlock()).toNumber(),
+    };
+    try {
+      const response = (await this.http.post("/execute", request))
+        .data as ExecuteResponse;
+
+      const outerExtrinsic = this.idn.tx.murmur.proxy(
+        response.payload.call_data.name,
+        response.payload.call_data.pos,
+        response.payload.call_data.commitment,
+        response.payload.call_data.ciphertext,
+        response.payload.call_data.proof_items,
+        response.payload.call_data.size,
+        extrinsic,
+      )
```

Comment:
It's really bad to send the extrinsic back and forth, this will be improved as part of https://github.com/ideal-lab5/murmur/issues/17